lib/set.rb

```diff
@@ -38,30 +36,128 @@
+#
+# First, what's elsewhere. \Set includes the module Enumerable,
+# which provides dozens of additional methods.
+#
+# In particular, this set does not have many methods of its own
+# for fetching or for iterating.
+# Instead, it relies on those in Enumerable.
+#
+# Here, class \Set provides methods that are useful for:
+#
+# - {Creating a Set}[#class-Set-label-Methods+for+Creating+a+Set]
+# - {Set Operations}[#class-Set-label-Methods+for+Set+Operations]
+# - {Comparing}[#class-Set-label-Methods+for+Comparing]
+# - {Querying}[#class-Set-label-Methods+for+Querying]
+# - {Assigning}[#class-Set-label-Methods+for+Assigning]
+# - {Deleting}[#class-Set-label-Methods+for+Deleting]
+# - {Converting}[#class-Set-label-Methods+for+Converting]
+# - {Iterating}[#class-Set-label-Methods+for+Iterating]
+# - {And more....}[#class-Set-label-Other+Methods]
+#
+# === Methods for Creating a \Set
+#
+# ::[]:: Returns a new set containing the given objects.
+# ::new:: Returns a new set containing either the given objects (if no block given)
+#         or the return values from the called block (if a block given).
+#
+# === Methods for Set Operations
+#
+# #union, #+, {|}[#method-i-7C]:: Returns a new set containing all elements from +self+
+#                                  and all elements from a given enumerable (no duplicates).
+# #intersection, {&}[#method-i-26]:: Returns a new set containing all elements common to +self+
+#                                     and a given enumerable.
+# #difference, #-:: Returns a copy of +self+ with all elements in a given enumerable removed.
+# {^}[#method-i-5E]:: Returns a new set containing all elements from +self+ and a given enumerable
+#                     except those common to both.
+#
+# === Methods for Comparing
+#
+# {<=>}[#method-i-3C-3D-3E]:: Returns -1, 0, or 1
+#                              as +self+ is less than, equal to, or greater than a given object.
+# {==}[#method-i-3D-3D]:: Returns whether +self+ and a given enumerable are equal, as determined by Object#eql?
+# #compare_by_identity?:: Returns whether the set considers only identity when comparing elements.
+#
+# === Methods for Querying
+#
+# #length, #size:: Returns the count of elements.
+# #empty?:: Returns whether the set has no elements.
+# #include?, #member?, #===:: Returns whether a given object is an element in the set.
+# #subset?, {<=}[#method-i-3C-3D]:: Returns whether a given object is a subset of the set.
+# #proper_subset?, {<}[#method-i-3C]:: Returns whether a given enumerable is a proper subset of the set.
+# #superset?, {<=}[#method-i-3E-3D]:: Returns whether a given enumerable is a superset of the set.
+# #proper_superset?, {>}[#method-i-3E]:: Returns whether a given enumerable is a proper superset of the set.
+# #disjoint?:: Returns +true+ if the set and a given enumerable have no common elements, +false+ otherwise.
+# #intersect?:: Returns +true+ if the set and a given enumerable have any common elements, +false+ othewise.
```

Comment:
Maybe these should be in comparing?

Reply:
Each returns boolean, thus query;ing each also does a comparision, thus comparing.  I thought (elsewhere) that some should be in more than one category, but @jeremyevans bridled at that.

Reply:
I distinguish the two by the type of argument they take, not by the return type.
Comparing: `foo(set) => bool`
Querying: `foo(obj) => bool`
In particular, `<=>` returns the same type of information as `<=`, `>=`, ...